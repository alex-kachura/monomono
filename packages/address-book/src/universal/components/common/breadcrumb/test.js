import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { renderProviders } from '@oneaccount/react-foundations/lib/test-utils';
import { DefaultThemeProvider } from '@beans/theme';
import Breadcrumb from '.';

function renderBreadcrumb(breadcrumb) {
  const providers = renderProviders({
    breadcrumb,
    initialData: {},
    appConfig: {
      region: 'GB',
      config: {
        GB: {
          externalApps: {
            tescoHomepage: 'mock-url',
          },
        },
      },
    },
    children: (
      <DefaultThemeProvider>
        <Breadcrumb />
      </DefaultThemeProvider>
    ),
  });

  return render(providers);
}

describe('Breadcrumb component', () => {
  afterEach(cleanup);

  describe('has breadcrumb', () => {
    it('should render correctly', () => {
      const { asFragment } = renderBreadcrumb([
        {
          href: 'mock-url',
          text: 'mock-text',
        },
      ]);
      const fragment = asFragment();

      expect(fragment).toMatchSnapshot();
    });
  });

  describe('no breadcrumb', () => {
    it('should render correctly', () => {
      const { asFragment } = renderBreadcrumb([]);
      const fragment = asFragment();

      expect(fragment).toMatchSnapshot();
    });
  });
});
