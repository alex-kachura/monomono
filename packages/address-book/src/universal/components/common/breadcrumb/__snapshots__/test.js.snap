// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component has breadcrumb should render correctly 1`] = `
<DocumentFragment>
  <div
    class="base-components__BaseElement-sc-150tvch-0 sc-bdVaJa dYnrJc"
  >
    <nav
      aria-label="breadcrumb"
      class="base-components__RootElement-sc-150tvch-1 styled__NavContainer-sc-1fzx5su-0 dXoXQG"
    >
      <ol
        class="base-components__BaseList-sc-150tvch-2 styled__OrderedList-sc-1fzx5su-2 dzvyLR"
      >
        <li
          class="styled__ListItem-sc-1fzx5su-3 kDICdn"
        >
          <div
            class="styled__Item-sc-1fzx5su-4 epQOWE"
          >
            <a
              aria-label="home"
              class="styled__IconLink-sc-117kvua-0 kZAYfa"
              href="mock-url"
            >
              <svg
                aria-hidden="true"
                class="styled__Container-sc-1bf1p22-0 gJOjrx"
                color="currentColor"
                height="22"
                preserveAspectRatio="xMinYMax meet"
                viewBox="0 1 20 22"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 1.5L1.5 8.25V20.5h6v-8h5v8h6V8.25z"
                  fill="none"
                  stroke="currentColor"
                />
              </svg>
            </a>
          </div>
        </li>
        <li
          class="styled__ListItem-sc-1fzx5su-3 kDICdn"
        >
          <div
            class="styled__Item-sc-1fzx5su-4 epQOWE"
          >
            <svg
              aria-hidden="true"
              class="styled__Separator-sc-1fzx5su-5 kaoddq styled__Container-sc-1bf1p22-0 gJOjrx"
              color="currentColor"
              height="20"
              preserveAspectRatio="xMinYMax meet"
              stroke="#666666"
              viewBox="-1 0 12 20"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.312 1.348L10.276 10l-8.964 8.644"
                fill="none"
                stroke="#666666"
              />
            </svg>
            <a
              class="styled__StandaloneLink-sc-1xizymv-0 IAUPG"
              href="mock-url"
            >
              <span
                class="beans-link__text styled__TextSpan-sc-1xizymv-3 hWdmzc"
              >
                mock-text
              </span>
            </a>
          </div>
        </li>
      </ol>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Breadcrumb component no breadcrumb should render correctly 1`] = `<DocumentFragment />`;
